pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/module/PublishDraftVersion.java

```diff
@@ -75,64 +84,74 @@ public void run(SlingHttpServletRequest request, PostResponse response, SlingPos
+        try (ResourceResolver serviceResourceResolver = serviceResourceResolverProvider.getServiceResourceResolver()) {
+            Module module = serviceResourceResolver.getResource(request.getResource().getPath()).adaptTo(Module.class);
+            Locale locale = getLocale(request);
+            String variant = getVariant(request);
 
-        // Get the draft version, there should be one
-        Optional<ModuleVersion> versionToRelease = module.getDraftVersion(locale, variant);
-        if( !versionToRelease.isPresent() ) {
-            response.setStatus(HttpServletResponse.SC_PRECONDITION_FAILED,
-                    "The module doesn't have a draft version to be released");
-        } else if (versionToRelease.get().metadata().getOrCreate().productVersion().get() == null
-        		||  versionToRelease.get().metadata().getOrCreate().productVersion().get().isEmpty()) {
-        	// Check if productVersion is set
-        	response.setStatus(HttpServletResponse.SC_PRECONDITION_FAILED,
-                    "The version to be released doesn't have productVersion metadata");
-        } else if (versionToRelease.get().metadata().getOrCreate().urlFragment().get() == null
-        		||  versionToRelease.get().metadata().getOrCreate().urlFragment().get().isEmpty()) {
-        	// Check if urlFragment is set
-        	response.setStatus(HttpServletResponse.SC_PRECONDITION_FAILED,
-                    "The version to be released doesn't have urlFragment metadata");
-        } else {
-            // Draft becomes the new released version
-            ModuleVariant moduleVariant = traverseFrom(module)
-                    .toChild(m -> module.locale(locale))
-                    .toChild(ModuleLocale::variants)
-                    .toChild(variants -> variants.variant(variant))
-                    .get();
-            moduleVariant.releaseDraft();
-            changes.add(Modification.onModified(module.getPath()));
-            // source/draft becomes source/released
-            FileResource draftSource = traverseFrom(module)
-                    .toChild(m -> module.locale(locale))
-                    .toChild(ModuleLocale::source)
-                    .toChild(sourceContent -> sourceContent.draft())
-                    .get();
-            // Check for released version
-            Optional<HashableFileResource> releasedSource = traverseFrom(module)
-                    .toChild(m -> module.locale(locale))
-                    .toChild(ModuleLocale::source)
-                    .toChild(sourceContent -> sourceContent.released())
-                    .getAsOptional();
-            if (draftSource != null) {
-                if (releasedSource.isPresent()) {
+            // Get the draft version, there should be one
+            Optional<ModuleVersion> versionToRelease = module.getDraftVersion(locale, variant);
+            if (!versionToRelease.isPresent()) {
```

Comment:
Rather than if/else if,  if and return  would be better as you don't want the process to go further after setting response.